hydromt/io/readers.py

```diff
@@ -862,9 +862,9 @@ def parse_values(
 
 
 def read_nc(
-    fn: StrPath,
-    root,
-    logger: Logger,
+    filename_template: StrPath,
```

Comment:
I disagree. I would argue that this is more accurate, as the template can specify multiple files, not just one. Could you explain a bit more why you think `fn` is more explicit?

Reply:
Indeed forgot we explicitly ask for {name} placeholder at least in ``write_nc`` to be in the path so filename_template sounds good.
For ``read_nc`` it's less clear though as it just says may contain wildcard and I think in main this just meant you can use * and the name is the basename of the file. Here is seems behavior may be different? Just we just be more explicit on a {name} placeholder like in write_nc ?

Reply:
well, I changed the behavior slightly since I wanted to make them more consistent. now `read_nc` actually also expects a `{name}` placeholder that will be used to determine the name of the dataset which will be used in the component. So now the same paths can be used for reading and writing. Therefore I think it's best to keep the names in both functions as well.